src/core/pointcloud/qgspointclouddataprovider.cpp

```diff
@@ -311,13 +311,36 @@ QVector<QVariantMap> QgsPointCloudDataProvider::identify(
   double maxError,
   const QgsGeometry &extentGeometry,
   const QgsDoubleRange &extentZRange, int pointsLimit )
+{
+  QVector<QVariantMap> acceptedPoints;
+
+  // Try sub-indexes first
+  for ( QgsPointCloudSubIndex &subidx : subIndexes() )
+  {
+    // Check if the sub-index is relevant and if it is loaded. We shouldn't
+    // need to identify points in unloaded indices.
+    if ( !subidx.index()
+         || !subidx.polygonBounds().intersects( extentGeometry )
+         || ( !subidx.zRange().overlaps( extentZRange ) ) )
```

Comment:
Let's do the (considerably!) cheaper check first:

```suggestion
         || ( !subidx.zRange().overlaps( extentZRange ) ) )
         || !subidx.polygonBounds().intersects( extentGeometry )
```

Reply:
Done.